test/unit/storage_async_test.rb

```diff
@@ -203,28 +203,29 @@ def test_tls_no_client_certificate
 
   [:rsa, :dsa, :ec].each do |alg|
     define_method "test_tls_client_cert_#{alg}" do
-      ca_file = create_cert
+      ca_cert_file = Tempfile.new('ca-root-cert.pem')
+      ca_cert_file.write(create_cert.to_pem)
+      ca_cert_file.rewind
```

Comment:
I'm not sure about your intention. Probably you mean `flush` and then `close` and `unlink` in an `ensure` block?

Reply:
I took that from your link (https://discuss.rubyonrails.org/t/validates-immutable-true/81479/2). I assumed `#rewind` was necessary to ensure the file is read from the beginning.

About unlinking, I'll add an ensure block.

Reply:
If you use the same file handle - yes. But you pass the files as paths. So they are opened again.

Reply:
Oh, true. I'll fix this.

Reply:
https://github.com/3scale/apisonator/pull/391/commits/c27f6c43a46bc21667d3c5830a6e5e764c0a610f